app/main.py

```diff
@@ -1 +1,24 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (NotVaccinatedError,
+                        NotWearingMaskError,
+                        OutdatedVaccineError
+                        )
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except (NotVaccinatedError, OutdatedVaccineError):
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_to_buy = sum(
+                1 for friend in friends if not friend["wearing_a_mask"]
+            )
```

Comment:
There is no need to use comprehension, you have already checked if friend has a mask in visit_cafe. Maybe it simpler will be to add one if we raise a NotWearingMaskError error